### PR TITLE
Make CoreBundle optional

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this changes are BC.

Part of https://github.com/sonata-project/dev-kit/issues/697

## Changelog

```markdown
### Fixed
- Fix wrong root node
### Removed
- Remove support for Symfony <4.4
- Remove SonataCoreBundle dependencies
```

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -117,10 +117,10 @@ public function getConfigTreeBuilder()
             ->end()
             ->scalarNode('slugify_service')
                 // NEXT_MAJOR: reword this info message
-                ->info('You should use: sonata.core.slugify.cocur, but for BC we keep \'sonata.core.slugify.native\' as default')
+                ->info('You should use: sonata.page.slugify.cocur, but for BC we keep \'sonata.core.slugify.native\' as default')
 
-                // NEXT_MAJOR: use "sonata.core.slugify.cocur" instead of "sonata.core.slugify.native" as default value
-                ->defaultValue('sonata.core.slugify.native')
```
Comment (jordisala1991) [inline src/DependencyInjection/Configuration.php:123]: Changing the default value would be a bc break, maybe register an alias to the new value on services?

File: src/SonataPageBundle.php [hunk 2]
```diff
@@ -85,17 +84,19 @@ public function boot()
     /**
      * Register form mapping information.
      */
-    public function registerFormMapping()
+    private function registerFormMapping()
```
Comment (jordisala1991) [inline src/SonataPageBundle.php:87]: Do not change visibility please

File: src/SonataPageBundle.php [hunk 3]
```diff
@@ -85,17 +84,19 @@ public function boot()
...
-            'sonata_page_api_form_site' => ApiSiteType::class,
-            'sonata_page_api_form_page' => ApiPageType::class,
-            'sonata_page_api_form_block' => ApiBlockType::class,
-            'sonata_page_selector' => PageSelectorType::class,
-            'sonata_page_create_snapshot' => CreateSnapshotType::class,
-            'sonata_page_template' => TemplateChoiceType::class,
-            'sonata_page_type_choice' => PageTypeChoiceType::class,
-            'cmf_routing_route_type' => RouteTypeType::class,
-        ]);
+        if (class_exists('Sonata\CoreBundle\Form\FormHelper')) {
```
Comment (jordisala1991) [inline src/SonataPageBundle.php:89]: FormHelper::class

File: composer.json [hunk 4]
```diff
@@ -48,17 +50,21 @@
         "symfony/routing": "^4.3",
         "symfony/security-core": "^4.3",
         "symfony/security-http": "^4.3",
+        "symfony/string": "^5.1",
         "symfony/templating": "^4.3",
         "symfony/validator": "^4.3",
-        "twig/twig": "^2.10"
+        "twig/extra-bundle": "^3.0",
```
Comment (jordisala1991) [inline composer.json:56]: I think this should be a separate PR

File: tests/DependencyInjection/SonataPageExtensionTest.php [hunk 5]
```diff
@@ -64,6 +64,8 @@ public function testRouterAutoRegister(): void
         $this->assertContainerBuilderHasParameter('sonata.page.router_auto_register.priority', 84);
     }
 
+    /*
+     * TODO: fix this test or remove it
     public function testDatePickerFormTheme(): void
```
Comment (wbloszyk) [inline tests/DependencyInjection/SonataPageExtensionTest.php:69]: @core23 @jordisala1991 @phansys Can you help me with this test? I can't fix it after check bundle exist in `SonataPageExtnesion`. Or mybe i should remove it?

Comment (wbloszyk) [reply tests/DependencyInjection/SonataPageExtensionTest.php]: Fixed

**Comment by wbloszyk** [comment]
@core23 @phansys @VincentLanglet @greg0ire 
It is almost RTM. In `--prefer-lowest` are some errors. It look like DI don't work propertly. Mybe you know why?

File: composer.json [hunk 6]
```diff
@@ -22,49 +22,52 @@
...
         "friendsofsymfony/rest-bundle": "<2.1 || >=3.0",
-        "jms/serializer": "<0.13"
+        "jms/serializer": "<0.13",
+        "sonata-project/core-bundle": "<3.20"
     },
     "require-dev": {
         "friendsofsymfony/rest-bundle": "^2.1",
-        "jms/serializer-bundle": "^1.0 || ^2.0",
+        "jms/serializer-bundle": "^2.0 || ^3.0",
         "matthiasnoback/symfony-dependency-injection-test": "^4.0",
```
Comment (greg0ire) [inline composer.json:66]: Bumping this one solves many issues. I tried `composer update --prefer-lowest` on this branch, and after updating it, I'm left with one issue.

Comment (greg0ire) [reply composer.json:66]: The other issue can be fixed by updating `doctrine/annotations`

Comment (greg0ire) [reply composer.json:66]: I've got some time, I'll do it for you :)

File: tests/App/AppKernel.php [hunk 7]
```diff
@@ -91,6 +100,6 @@ protected function configureContainer(ContainerBuilder $containerBuilder, Loader
 
     private function getBaseDir(): string
     {
-        return sys_get_temp_dir().'/sonata-page-bundle/var/';
+        return sys_get_temp_dir().'/sonata-page-bundle---5353535a/var/';
```
Comment (jordisala1991) [inline tests/App/AppKernel.php:103]: this should not be changed

File: composer.json (hunk 6 above)
Comment (wbloszyk) [reply composer.json]: @greg0ire Thanks. I do it in other bundles but here i miss it.

File: tests/App/AppKernel.php (hunk 7 above)
Comment (wbloszyk) [reply tests/App/AppKernel.php]: I know about this. I must change change this param each time I make functional test. I dont change it to correct becose I had some problem. @greg0ire  fix it.

**Comment by wbloszyk** [comment]
It is finnaly RTM.

**Comment by jordisala1991** [comment]
Thank you @wbloszyk
